resources-domain/packages/resources-search-config/resourceIndexDocumentMappings.ts

```diff
@@ -41,8 +41,12 @@ export type FieldAndMapping = {
 
 const stringFieldTypes = ['text', 'keyword'];
 
-export const isMappingField = ({ mappingFields }: Pick<ResourceIndexDocumentMappings, 'mappingFields'>, fieldName: string) =>
-  Object.keys(mappingFields).includes(fieldName);
+export const getMappingField = ({ mappingFields }: Pick<ResourceIndexDocumentMappings, 'mappingFields'>, fieldName: string): FieldAndMapping | undefined => {
+  if (Object.keys(mappingFields).includes(fieldName)) return { field: fieldName, mapping:mappingFields[fieldName] };
+  const [field, mapping] = Object.entries(mappingFields).find(([, { attributeKeyPattern }]) => attributeKeyPattern?.test(fieldName)) ?? [];
+  return mapping && field ? { field, mapping } : undefined;
+};
```

Comment:
This one is short but I feel like does a lot. Can you explain to me what's going on here? :P

Reply:
I added a comment explaining the workflow, figured if you had issues following it, others would too